libecho/src/RawReceiver.cc

```diff
@@ -0,0 +1,108 @@
+#include <cmath>
+#include "RawReceiver.h"
+
+#define SAMPLE_RATE 44100
+
+static double dft(const int16_t *buffer, int samples, double ratio) {
+    double re = 0;
+    double im = 0;
+    double angle = 0;
+    const double d_angle = 2.0 * M_PI * ratio;
+
+    for (int i = 0; i < samples; i++) {
+        double val = buffer[i];
+        re += val * std::cos(angle);
+        im += val * std::sin(angle);
+        angle += d_angle;
+    }
+
+    return std::sqrt(re * re + im * im) / samples;
+}
+
+RawReceiver::RawReceiver(int loFreq, int hiFreq, double magLimitBegin, double magLimitEnd, int winSize) {
+    format.setByteOrder(QAudioFormat::LittleEndian);
+    format.setChannelCount(1);
+    format.setCodec("audio/pcm");
+    format.setSampleRate(SAMPLE_RATE);
+    format.setSampleSize(16);
+    format.setSampleType(QAudioFormat::SignedInt);
```

Comment:
Maybe move that to the helper method and define arguments as const expressions?

Reply:
I mean, something like:
format = getInputFormat();

Reply:
i don't see how that's a helper method, it doesn't help with anything